### DISCUSSION TITLE
【国际化】在React组件中可以使用FormattedMessage，在非react的ts文件中如何使用国际化呢

### DISCUSSION THREAD

**Discussion by deronendless** [umijs/umi#12414] [Q&A]
在React组件中可以使用FormattedMessage，在非react的ts文件中如何使用国际化呢
比如我再constants.ts文件中，如何使用呢？

**Comment by fz6m** [comment]
如果不在 hooks 里，就用 `getIntl` 拿到国际化 intl 实例再去调用国际化方法，但你要确保你使用 intl 的时机不是在全局的，而是在应用启动后的某个时机，因为全局使用的时候还没加载到 i18n 的 context ，国际化还没启动起来，所以必须要在应用启动之后的时机去调用，往往在 hooks 、函数里、组件里去使用，要延后使用，直接在全局使用很可能会获取不到内容。
